server/lyft/command/feature_runner.go

```diff
@@ -0,0 +1,63 @@
+package command
+
+import (
+	"github.com/runatlantis/atlantis/server/events"
+	"github.com/runatlantis/atlantis/server/events/command"
+	"github.com/runatlantis/atlantis/server/logging"
+	"github.com/runatlantis/atlantis/server/lyft/feature"
+)
+
+func NewPlatformModeFeatureRunner(
+	featureAllocator feature.Allocator,
+	platformModeEnabled bool,
+	logger logging.SimpleLogging,
+) *PlatformModeFeatureRunner {
+	return &PlatformModeFeatureRunner{
+		featureAllocator:    featureAllocator,
+		platformModeEnabled: platformModeEnabled,
+		logger:              logger,
+	}
+}
+
+// PlatformModeFeatureRunner basic struct to that prepares a runner based on if
+// the platform mode feature is enabled for the repo and if the server is
+// running in platform mode
+type PlatformModeFeatureRunner struct {
+	featureAllocator    feature.Allocator
+	platformModeEnabled bool
+	logger              logging.SimpleLogging
+}
+
+type featureRunnerFunc func(ctx *command.Context, cmd *events.CommentCommand)
+
+func (f featureRunnerFunc) Run(ctx *command.Context, cmd *events.CommentCommand) {
+	f(ctx, cmd)
+}
```

Comment:
Why is this necessary vs. just an interface? All the methods seem to have the same signature?  Is there a reason we can't just use a decorator?

Reply:
This is to avoid instantiating a new decorator instance for each command. For instance in `server.go` I'm doing this:
```
	featuredRunner := lyftCommands.NewPlatformModeFeatureRunner(
		featureAllocator,
		userConfig.EnablePlatformMode,
		logger,
	)

	commentCommandRunnerByCmd := map[command.Name]comment.Runner{
		command.Plan: featuredRunner.Wrap(
			plan.NewRunner(vcsClient),
			planCommandRunner,
		),
		command.Apply: featuredRunner.Wrap(
			apply.NewRunner(vcsClient),
			applyCommandRunner,
		),
		command.ApprovePolicies: featuredRunner.Wrap(
			policies.NewRunner(vcsClient),
			approvePoliciesCommandRunner,
		)
	}
```
Instead of initializing a new runner for each plan. This is basically a decorator pattern as well, just more functional inspired. Based on [this](https://www.youtube.com/watch?v=xyDkyFjzFVc) talk. I also fixed the signature of the `Wrap` function to return `CommentCommandRunner` interface.

Reply:
I changed the code to use decorator struct.